.github/workflows/aws-deploy.yaml

```diff
@@ -0,0 +1,70 @@
+name: aws-deploy
+
+on: push
+
+jobs:
+  pre-commit:
+    name: Run pre-commit hooks against all files
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: pre-commit/action@v2.0.2
+
+  sceptre-deploy:
+    name: Deploy CloudFormation templates using sceptre
+    runs-on: ubuntu-latest
+    needs: pre-commit
+    if: github.ref == 'refs/heads/main'
+    strategy:
+      matrix:
+        job-environment:
+          - develop
+          - prod
+        include:
+          - job-environment: develop
+            sceptre-environment: develop
+          - job-environment: prod
+            sceptre-environment: prod
+    environment: ${{ matrix.job-environment }}
```

Comment:
Lines 19-29 here are a new pattern; I introduced this to reduce duplication when deploying to more than one environment (as we currently have in Sage-Bionetworks-Workflows/aws-workflows-nextflow-infra repo). This depends upon there being a "develop" and a "prod" Environment in the settings for this github repo, containing some environment-specific secrets.

I chose to keep "job-environment" and "sceptre-environment" separate for a couple reasons: first, they're actually different, that is github vs sceptre concepts, and so can diverge. If we conflate them now and someone decides the sceptre "develop" folder should really be called "dev", that will break the CICD. Second, I was testing out defining additional variables in matrix includes. That said I could definitely be convinced to use the following instead:
```
strategy:
  matrix:
    environment:
      - develop
      - prod
```